### PR TITLE
[dnsmasq] Update patch to adjust for CVE in upstream

### PR DESCRIPTION
**Pull request by crogers1**
dnsmasq recipe

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

### CODE REVIEW

File: recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch [hunk 1]
```diff
@@ -102,3 +75,31 @@
...
++      strncpy(ifr.ifr_name, ifn->name, sizeof (ifr.ifr_name));
+ 
+   if ((fd = socket(s->source_addr.sa.sa_family, SOCK_DGRAM, 0)) != -1)
+     {
++            if (ifr.ifr_name[0])
++            {
++                printf("Use interface %s\n", ifr.ifr_name);
++                if (setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE,
++                               (void *)&ifr, sizeof (ifr)) < 0) 
++			    {
```
Comment (jandryuk) [inline recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch:98]: Generally looks good, but you let some tabs slip in here while the rest of the file is spaces.

**Review by jandryuk — CHANGES_REQUESTED** [review]

File: recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch (hunk 1 above)
Comment (crogers1) [reply recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch:98]: The source file (forward.c) seems to be mixed tabs and spaced.  Usually 1-2 tabs depending on indent level, then spaces to align.  it's pretty awful.

Comment (crogers1) [reply recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch:98]: I could update the patch to only use spaces but technically it wouldn't follow what forward.c is doing elsewhere

Comment (jandryuk) [reply recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch:98]: It also looks like the meta-oe backport https://github.com/openembedded/meta-openembedded/blob/2e7e98cd0cb82db214b13224c71134b9335a719b/meta-networking/recipes-support/dnsmasq/dnsmasq/CVE-2021-3448.patch made a mess of spaces and tabs.

Still, I think our addition should at least be internally consistent.  So do your best?

Comment (crogers1) [reply recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch:98]: ack, one sec

**Comment by crogers1** [comment]
Should be resolved with the latest push

**Review by jandryuk — APPROVED** [review]
LGTM, thanks

**Comment by jandryuk** [comment]
Will merge soon.

**Comment by crogers1** [comment]
Build here: [build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/55)

**Pull request merged**